fontforge/splinefont.c

```diff
@@ -931,14 +931,37 @@ return(copy(tmpfilename));			/* The filename does not exist */
     }
 }
 
+/* The following won't work if the fontname has unbalanced
+ * parentheses, but short of requiring those to be escaped
+ * somehow that case is tricky. (One could guess the cutoff
+ * by looking for a file extension.)
+ * */
+char *SFSubfontnameStart(char *fname) {
+    char *rparen, *paren;
+    int cnt = 1;
+    if ( fname==NULL || (rparen = strrchr(fname,')'))==NULL || rparen[1]!='\0' )
```

Comment:
So fname is invalid if the closing parenthesis occurs before the last character? Why is that?

Reply:
I added a comment that explains this, which was a preexisting convention. The code had been:
```
    if ( (paren = strrchr(pt,'('))!=NULL &&
	    (rparen = strrchr(paren,')'))!=NULL &&
	    rparen[1]=='\0' ) {
```

The restriction that the subfontname come at the end of the string likely corresponds to the possibility of a font filename containing parentheses, given that they are valid characters in most filesystems.

Reply:
> The restriction that the subfontname come at the end of the string likely corresponds to the possibility of a font filename containing parentheses, given that they are valid characters in most filesystems.

Filenames should never contain parentheses because the file operating systems sees this as a "list".
The fontforge open command looks at this differently as mentioned in #3133 
```code
fontforge.open(file_name + '(' + font_name + ')')
```
file_name == 'Automatica -BRK-xy.ttf' and font_name == 'Automatica (BRK)xy'

Reply:
> Filenames should never contain parentheses because the file operating systems sees this as a "list".

What, that doesn't make sense, the file system doesn't care, it's your bash/shell interpreter that's doing that. Escape properly and you can use braces all you like.